### DISCUSSION TITLE
Get hub address...

### DISCUSSION THREAD

**Discussion by vascolp** [pybricks/support#497] [Q&A]
Hi,

Is there a simple way to get the mac address on a TechnicHub or CityHub object?

Thanks and Regards,
vascolp

**Comment by vascolp** [comment]
I mean, inside python code running on it....

**Comment by dlech** [comment]
No there isn't currently a way to do this. Why do you need the address?

**Comment by vascolp** [comment]
Well...
I want a way to store some parameters in the Hub, when I turn it off, so that I have the same configuration when turned on.
One good place could be the name of the Hub, but there is no way to set it yet...
However, I noticed I could use the Remote name property, because I can set it.... and it would be nice to have it configured for a given hub only.... the mac address is a good unique id for the hub.
Thanks for your fast answer!

**Comment by laurensvalk** [comment]
> I want a way to store some parameters in the Hub, when I turn it off, so that I have the same configuration when turned on.

You can save a program with parameters on the hub.

For example, save this program on the hub:

```python



my_data = "Test"
more_data = (1, 2, 3)
```

Now you can run a new program that uses this information:

```python

from main import my_data, more_data

print(my_data)
print(more_data)

```

**Comment by vascolp** [comment]
Hmmmm :-) 
Is that read-write? Can I change them? For instance 
     my_data="1234"
and get 1234 next time I run?

**Reply by laurensvalk** [reply]
No, but that wouldn't be the case for the MAC address either.

These hubs do not really have storage space for user data. There's only the firmware itself.

**Reply by vascolp** [reply]
I know MAC is read-only, and MAC was indid was my first question...
What I would like is to be able to save a small amout of data on the hub, persistently. The 14 charecters from the name would be enough.
Currently name on the hub is not read-write , so I will use name from the remote, which is writable and persistent even if you remove batteries. 
I just thought I could tie that config in the remote with the given hub by storing hub MAC address on the remote also..... 
but I live without it. In fact beeing able to change the Hub name is much more interesting than getting its MAC address :-)
Thank you!
